tests/cli/test_build.py

```diff
@@ -0,0 +1,143 @@
+#
+# Copyright (C) 2020 Arm Mbed. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+import contextlib
+import os
+import pathlib
+
+from tempfile import TemporaryDirectory
+from unittest import TestCase, mock
+
+from click.testing import CliRunner
+
+from mbed_tools.cli.build import build
+
+
+@contextlib.contextmanager
+def mock_project_directory(mbed_config_exists=False, build_tree_exists=False):
+    with TemporaryDirectory() as tmp_dir:
+        root = pathlib.Path(tmp_dir, "test-project")
+        root.mkdir()
+        build_tree = root / "cmake_build"
+        mbed_config_dir = root / ".mbedbuild"
+        mbed_config_file = mbed_config_dir / "mbed_config.cmake"
```

Comment:
Would tests that use this mock stop working if we renamed the config file? Should we query the path from a module that owns the path here instead? It looks like this mock is "by chance" matching what the other module uses, but that could change if the module changes.